src/search/mod.rs

```diff
@@ -10,24 +13,51 @@ pub mod indexing;
 
 #[derive(Error, Debug)]
 pub enum SearchError {
-    #[error("Error while connection to the MeiliSearch database")]
-    IndexDBError(),
-    #[error("Error while connecting to the local server")]
-    LocalDatabaseError(#[from] mongodb::error::Error),
-    #[error("Error while accessing the data from remote")]
-    RemoteWebsiteError(#[from] reqwest::Error),
-    #[error("Error while serializing or deserializing JSON")]
+    #[error("Error while connecting to the MeiliSearch database")]
+    IndexDBError(meilisearch_sdk::errors::Error),
+    #[error("Error while importing mods from CurseForge")]
+    CurseforgeImportError(reqwest::Error),
+    #[error("Error while serializing or deserializing JSON: {0}")]
     SerDeError(#[from] serde_json::Error),
-    #[error("Error while parsing float")]
+    #[error("Error while parsing a float: {0}")]
     FloatParsingError(#[from] std::num::ParseFloatError),
-    #[error("Error while parsing float")]
+    #[error("Error while parsing an integer: {0}")]
     IntParsingError(#[from] std::num::ParseIntError),
-    #[error("Error while parsing BSON")]
+    #[error("Database Error: {0}")]
     DatabaseError(#[from] DatabaseError),
     #[error("Environment Error")]
     EnvError(#[from] dotenv::Error),
 }
 
+impl actix_web::ResponseError for SearchError {
+    fn status_code(&self) -> StatusCode {
+        match self {
+            SearchError::EnvError(..) => StatusCode::INTERNAL_SERVER_ERROR,
+            SearchError::DatabaseError(..) => StatusCode::INTERNAL_SERVER_ERROR,
+            SearchError::IndexDBError(..) => StatusCode::INTERNAL_SERVER_ERROR,
+            SearchError::CurseforgeImportError(..) => StatusCode::INTERNAL_SERVER_ERROR,
+            SearchError::SerDeError(..) => StatusCode::BAD_REQUEST,
+            SearchError::FloatParsingError(..) => StatusCode::BAD_REQUEST,
+            SearchError::IntParsingError(..) => StatusCode::BAD_REQUEST,
+        }
+    }
+
+    fn error_response(&self) -> HttpResponse {
```

Comment:
The json `error` should be a snake_case message that is consistent between endpoints